jetty-websocket/websocket-client/src/main/java/org/eclipse/jetty/websocket/client/WebSocketClient.java

```diff
@@ -112,11 +116,13 @@ public WebSocketClient(HttpClient httpClient)
      */
     public WebSocketClient(HttpClient httpClient, DecoratedObjectFactory objectFactory)
     {
-        this.containerScope = new SimpleContainerScope(WebSocketPolicy.newClientPolicy(),new MappedByteBufferPool(),objectFactory);
-        this.httpClient = httpClient;
-        this.extensionRegistry = new WebSocketExtensionFactory(containerScope);
-        this.eventDriverFactory = new EventDriverFactory(containerScope);
-        this.sessionFactory = new WebSocketSessionFactory(containerScope);
+        this.httpClient = Objects.requireNonNull(httpClient, "HttpClient");
+        this.policy = new WebSocketPolicy(WebSocketBehavior.CLIENT);
+        final DecoratedObjectFactory decoratedObjectFactory = objectFactory != null ? objectFactory : newDecoratedObjectFactory();
+        this.objectFactorySupplier = () -> decoratedObjectFactory;
+        this.extensionRegistry = new WebSocketExtensionFactory(this);
+        this.eventDriverFactory = new EventDriverFactory(this);
+        this.sessionFactory = new WebSocketSessionFactory(this);
```

Comment:
Would it be possible to have just one constructor that assigns fields and all the other constructors delegating to this one.
The field-assigning constructor may be private with many parameters if needs to be.